client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -21,153 +21,307 @@ import SetupHeader from './setup-header';
+				} ).then( () => {
+					this.props.fetchPlugins( [ site.ID ] );
+				} );
+			} else {
+				// Otherwise, handle plugin installation the normal way
+				this.props.installPlugin( site.ID, getPlugin( wporg, workingOn ) );
+			}
+
+			this.setState( {
+				message: translate( 'Installing %(plugin)s', { args: { plugin: requiredPlugins[ workingOn ] } } ),
+				toInstall,
+				workingOn,
+			} );
+			return;
+		}
+
+		// Otherwise, if we are working on something presently, see if it has appeared in state yet
+		const pluginFound = find( sitePlugins, { slug: this.state.workingOn } );
+		if ( pluginFound ) {
+			this.setState( {
+				workingOn: '',
+			} );
+		}
 	}
 
-	installPlugins = ( plugins ) => {
-		this.cancelUpdateTimeout();
-		const { site, wporg } = this.props;
-		for ( let i = 0; i < requiredPlugins.length; i++ ) {
-			const slug = requiredPlugins[ i ];
-			const plugin = find( plugins, { slug } );
-
-			if ( ! plugin ) {
-				if ( 'wc-api-dev' === slug ) {
-					if ( 'wc-api-dev' !== this.state.installingPlugin ) {
-						this.installApiDevPlugin( site.ID );
-					}
-
-					return;
-				}
-
-				if ( ! wporg[ slug ] ) {
-					return;
-				}
-
-				const wporgPlugin = getPlugin( wporg, slug );
-				const progress = this.state.progress + ( 100 / requiredPlugins.length );
-				this.setState( () => ( {
-					installingPlugin: slug,
-					progress,
-				} ) );
-				this.props.installPlugin( site.ID, wporgPlugin );
-				this.setUpdateTimeout();
+	doActivation = () => {
+		const { site, sitePlugins, translate } = this.props;
+		const requiredPlugins = this.getRequiredPluginsList();
+
+		// If we are working on nothing presently, get the next thing to activate and activate it
+		if ( 0 === this.state.workingOn.length ) {
+			const toActivate = this.state.toActivate;
+
+			// Nothing left to activate? Advance to done success
+			if ( 0 === toActivate.length ) {
+				this.setState( {
+					engineState: 'DONESUCCESS',
+					message: '',
+					progress: 100,
+				} );
 				return;
 			}
-			if ( ! plugin.active ) {
-				const wporgPlugin = getPlugin( wporg, slug );
-				this.setState( () => ( { activatingPlugin: slug } ) );
-				this.props.activatePlugin( site.ID, { ...wporgPlugin, id: plugin.id } );
-				this.setUpdateTimeout();
+
+			const workingOn = toActivate.shift();
+
+			// It is best to use sitePlugins to get the right id since the
+			// plugin id isn't always slug/slug unless the main plugin PHP
+			// file is the same name as the plugin folder
+			const pluginToActivate = find( sitePlugins, { slug: workingOn } );
+			// Already active? Skip it
+			if ( pluginToActivate.active ) {
+				this.setState( {
+					toActivate,
+					workingOn: '',
+				} );
 				return;
 			}
+
+			// Otherwise, activate!
+			this.props.activatePlugin( site.ID, pluginToActivate );
+
+			this.setState( {
+				message: translate( 'Activating %(plugin)s', { args: { plugin: requiredPlugins[ workingOn ] } } ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Activating %(plugin)s…' )` *ES Score: 20*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).